cuda/src/memory/cuda/resources/terminal/device_resource.cpp

```diff
@@ -0,0 +1,36 @@
+#include <memory>
+
+#include <cuda_runtime_api.h>
+
+#include "vecmem/memory/resources/memory_resource.hpp"
+#include "vecmem/memory/cuda/resources/terminal/device_resource.hpp"
+#include "vecmem/utils/cuda_error_handling.hpp"
```

Comment:
We don't have any coding guidelines for the project yet, but let's just adopt the ATLAS one. I.e. include stuff in the following order:
  - Project headers;
  - Headers from external projects;
  - System headers.

So, going from the most specific to the least specific. This tends to be a good way of detecting problems in one's header files.